data-processing/common/commands/fetch_s2_data.py

```diff
@@ -102,10 +105,20 @@ def process(self, item: ArxivId) -> Iterator[S2Metadata]:
                     authors = []
                     for author_data in reference_data["authors"]:
                         authors.append(Author(author_data["authorId"], author_data["name"]))
+
+                    external_ids = reference_data["externalIds"]
+                    if external_ids:
+                        corpus_id = external_ids.get("CorpusId")
+                        arxiv_id = external_ids.get("ArXiv")
+                        doi = external_ids.get("DOI")
+                    else:
+                        corpus_id = arxiv_id = doi = None
+
                     reference = Reference(
                         s2_id=reference_data["paperId"],
-                        arxivId=reference_data["arxivId"],
-                        doi=reference_data["doi"],
+                        corpus_id=corpus_id,
```

Comment:
Small suggestion: maybe let's put corpus_id at the end, in case there's anything that relies on the order that these fields appear in? Having said that, if you're confident you know all the usages, and there's nothing that does rely on the order that would get messed up, it's probably find to do it this way...

Reply:
oh yeah, i only put that in that order so I can keep `corpus_id` close to `s2_id` so it's easier on the eyeballs.  I think it'll be fine to leave as is though.